File: 1-easy/problems/2399-check-distances-between-same-letters/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def checkDistances(self, s, distance):
        """
        :type s: str
        :type distance: List[int]
        :rtype: bool
        """
        for i in range(len(s)):
            if i+distance[ord(s[i])-ord('a')]+1 >= len(s) or s[i+distance[ord(s[i])-ord('a')]+1] != s[i]:
                return False
            distance[ord(s[i])-ord('a')] = -1
        return True

File: 1-easy/problems/2399-check-distances-between-same-letters/solution.ts
function checkDistances(s: any, distance: any): boolean | number | string | any {
  """
  :type s: str
  :type distance: List[int]
  :rtype: bool
  """
  for (i in Array.from({length: (s}, (_, i) => i).length)) {
      if (i+distance[ord(s[i])-ord('a')]+1 >= (s).length || s[i+distance[ord(s[i])-ord('a')]+1] != s[i]) {
          return false;
      distance[ord(s[i])-ord('a')] = -1
  return true;
}

export { checkDistances };
